File: javaSrc/src/ice/type/IceLiquid.java
package ice.type;

import ice.ui.stat.IceStat;
import ice.ui.stat.IceStats;
import arc.graphics.Color;
import mindustry.type.Liquid;

public class IceLiquid extends Liquid {
    public float nutrientConcentration=0;
    public IceLiquid(String name, Color color) {
        super(name, color);
    }
    public IceLiquid(String name, String color) {
        super(name, Color.valueOf(color));
    }
    public IceLiquid(String name) {
        this(name,new Color(Color.black));
    }
    @Override
    public void init() {
        stats = new IceStats();
        super.init();
    }
    @Override
    public void setStats() {
        IceStats stats1 = (IceStats) stats;
        if (nutrientConcentration != 0) {
            stats1.addPercentThrob(IceStat.nutrientConcentration, nutrientConcentration, "[red]");
        }
        super.setStats();
    }
}

File: src/ice/library/type/baseContent/liquid/IceLiquid.kt
package ice.library.type.baseContent.liquid

import arc.graphics.Color
import ice.library.type.components.NutrientConcentration
import ice.library.type.meta.stat.IceStats
import ice.library.type.meta.stat.Stats
import mindustry.type.Liquid

open class IceLiquid(name: String, color: String) : Liquid(name, Color.valueOf(color)), NutrientConcentration {

    override var nutrientConcentration: Float = 0f

    override fun init() {
        stats = Stats()
        super.init()
    }

    override fun setStats() {
        val stats1 = stats as Stats
        if (nutrientConcentration != 0f) stats1.addPercent(IceStats.营养浓度, nutrientConcentration)
        super.setStats()
    }

}
